### DISCUSSION TITLE
Unable to add keys to Microsoft account

### DISCUSSION THREAD

**Discussion by huaracheguarache** [solokeys/solo2#113] [Q&A]
When I try to add my Solo 2 keys to my Microsoft account I get the following error message: "This security key can't be used. Please try a different one." This happens after I'm asked to create a PIN for the key during setup. Both of my keys are running the latest firmware: 1:20200101.9

**Comment by huaracheguarache** [comment]
@nickray Are you able to use the key on Windows/with Microsoft accounts? I'm sort of frustrated to see an issue like this after having to wait such a long time to get my key due to the delays with production. I was hoping these things would have been ironed out by now.

**Reply by huaracheguarache** [reply]
Ping @conorpp.

**Reply by huaracheguarache** [reply]
Any update on this issue? The lack of response from the developers together with the lack of documentation about how to update the firmware and use the key in general is pretty worrying.

**Reply by imilchev** [reply]
running into the same issue with Keeper. I have no clue how to update the firmware and nobody is replying to questions about why the keys generated by the device are longer than they are supposed to...

**Reply by dulfe** [reply]
I also have issues adding the key to Windows Account, after many tries I was able to do it, but using it also takes a lot of trial-and-errors.

The support team replied to my email and told me that a firmware fix should be coming in the "next few weeks", so, I have to wait-and-see if that fixes the issue.

About updating the firmware, you should use [solo2-cli](https://github.com/solokeys/solo2-cli), I installed it and it I was able to upgrade the firmware using this command:

`solo2 update`

That downloaded and installed the firmware version 1:20200101.9

Hope it helps.

**Comment by weiss-johannes** [comment]
Unplugging and plugging back in worked for me. Unfortunately, I have to do this every time I log on somewhere. The touch keys have never worked.
